src/pipeline_factory.hpp

```diff
@@ -56,6 +56,16 @@ class PipelineFactory {
         const tensorflow::serving::PredictRequest* request,
         tensorflow::serving::PredictResponse* response,
         ModelManager& manager) const;
+
+    PipelineDefinition* findDefinitionByName(const std::string& name) const {
```

Comment:
Why not reference?

Reply:
What if there is no definition with such name?   We might call ::definitionExists(name) before but that would require accessing definition map twice.